CodeSnippetsReflection.OpenAPI.Test/SnippetCodeGraphTests.cs

```diff
@@ -0,0 +1,310 @@
+
+            Assert.Equal(expectedObject.Name, result.Body.Name);
+            Assert.Equal(expectedObject.Value, result.Body.Value);
+            Assert.Equal(expectedObject.PropertyType, result.Body.PropertyType);
+        }
+
+        private CodeProperty? findProperyInSnipet(CodeProperty codeProperty, string name)
+        {
+            if (codeProperty.Name == name) return codeProperty;
+
+            if (codeProperty.Children.Any())
+            {
+                foreach (var param in codeProperty.Children)
+                {
+                    if(findProperyInSnipet(param, name) is CodeProperty result) return result;
+                }
+            }
+
+            return null;
+        }
+
+        [Fact]
+        public async Task ParsesBodyPropertyTypeString()
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Post, $"{ServiceRootUrl}/me/findMeetingTimes")
+            {
+                Content = new StringContent(TypesSample, Encoding.UTF8)
+            };
+            var snippetModel = new SnippetModel(request, ServiceRootUrl, await GetV1TreeNode());
+            var snippetCodeGraph = new SnippetCodeGraph(snippetModel);
+
+            // meetingDuration should be a string
+            var property = findProperyInSnipet(snippetCodeGraph.Body, "meetingDuration");
+
+            Assert.NotNull(property);
+            Assert.Equal(PropertyType.String, property?.PropertyType);
+            Assert.Equal("PT1H", property?.Value);
+        }
+
+        [Fact]
+        public async Task ParsesBodyPropertyTypeNumber()
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Post, $"{ServiceRootUrl}/me/findMeetingTimes")
+            {
+                Content = new StringContent(TypesSample, Encoding.UTF8)
+            };
+            var snippetModel = new SnippetModel(request, ServiceRootUrl, await GetV1TreeNode());
+            var snippetCodeGraph = new SnippetCodeGraph(snippetModel);
+
+            var property = findProperyInSnipet(snippetCodeGraph.Body, "minimumAttendeePercentage");
+
+            Assert.NotNull(property);
+            Assert.Equal(PropertyType.Number, property?.PropertyType);
+            Assert.Equal("100" , property?.Value);
+        }
+
+        [Fact]
+        public async Task ParsesBodyPropertyTypeBoolean()
+        {
+            using var request = new HttpRequestMessage(HttpMethod.Post, $"{ServiceRootUrl}/me/findMeetingTimes")
+            {
+                Content = new StringContent(TypesSample, Encoding.UTF8)
+            };
+            var snippetModel = new SnippetModel(request, ServiceRootUrl, await GetV1TreeNode());
+            var snippetCodeGraph = new SnippetCodeGraph(snippetModel);
+
+            var property = findProperyInSnipet(snippetCodeGraph.Body, "suggestLocation");
+
+            Assert.NotNull(property);
+            Assert.Equal(PropertyType.Boolean, property?.PropertyType);
+            Assert.Equal("False", property?.Value);
```

Comment:
guaranteed to be not null in these lines.
```suggestion
            Assert.Equal(PropertyType.Boolean, property.PropertyType);
            Assert.Equal("False", property.Value);
```